File: services/intelligence/common/logging.py
"""统一JSON结构化日志"""
import json
import sys
import time
import uuid
from datetime import datetime, timezone


def log(level: str, service: str, message: str, trace_id: str = "", **detail):
    entry = {
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "level": level,
        "service": service,
        "trace_id": trace_id,
        "message": message,
    }
    if detail:
        entry["detail"] = detail
    print(json.dumps(entry, ensure_ascii=False), file=sys.stderr, flush=True)


def info(service: str, message: str, trace_id: str = "", **detail):
    log("INFO", service, message, trace_id, **detail)


def warn(service: str, message: str, trace_id: str = "", **detail):
    log("WARN", service, message, trace_id, **detail)


def error(service: str, message: str, trace_id: str = "", **detail):
    log("ERROR", service, message, trace_id, **detail)

File: services/rule-extractor/src/common/logging.ts
const SERVICE = "rule-extractor";

type LogLevel = "ERROR" | "WARN" | "INFO" | "DEBUG";

interface LogEntry {
  timestamp: string;
  level: LogLevel;
  service: string;
  trace_id: string;
  message: string;
  detail?: Record<string, unknown>;
}

function write(level: LogLevel, traceId: string, message: string, detail?: Record<string, unknown>): void {
  const entry: LogEntry = {
    timestamp: new Date().toISOString(),
    level,
    service: SERVICE,
    trace_id: traceId,
    message,
  };
  if (detail && Object.keys(detail).length > 0) {
    entry.detail = detail;
  }
  process.stderr.write(JSON.stringify(entry) + "\n");
}

export const log = {
  info: (traceId: string, message: string, detail?: Record<string, unknown>) =>
    write("INFO", traceId, message, detail),
  warn: (traceId: string, message: string, detail?: Record<string, unknown>) =>
    write("WARN", traceId, message, detail),
  error: (traceId: string, message: string, detail?: Record<string, unknown>) =>
    write("ERROR", traceId, message, detail),
  debug: (traceId: string, message: string, detail?: Record<string, unknown>) =>
    write("DEBUG", traceId, message, detail),
};
